src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -1185,14 +1185,18 @@ private void ReportInvalidInstruction(ILOpcode opcode)
             ThrowHelper.ThrowInvalidProgramException();
         }
 
-        private bool IsRuntimeHelpersInitializeArray(MethodDesc method)
+        private bool IsRuntimeHelpersInitializeArrayOrGetSpanData(MethodDesc method)
         {
-            if (method.IsIntrinsic && method.Name == "InitializeArray")
+            if (method.IsIntrinsic)
             {
-                MetadataType owningType = method.OwningType as MetadataType;
-                if (owningType != null)
+                string name = method.Name;
+                if (name == "InitializeArray" || name == "GetSpanData")
```

Comment:
I think we need to look for CreateSpan instead - it's the one marked as intrinsic and the one where the ldtoken/call pattern will show up.